portal-impl/src/com/liferay/portal/service/impl/PhoneServiceImpl.java

```diff
@@ -63,6 +64,23 @@ public void deletePhone(long phoneId) throws PortalException {
 		phoneLocalService.deletePhone(phone);
 	}
 
+	@Override
+	public Phone fetchPhoneByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		Phone phone = phoneLocalService.fetchPhoneByExternalReferenceCode(
+			externalReferenceCode, companyId);
+
+		if (phone != null) {
+			CommonPermissionUtil.check(
+				getPermissionChecker(), phone.getClassNameId(),
+				phone.getClassPK(), ActionKeys.VIEW);
+		}
+
+		return phone;
+	}
+
```

Comment:
can we add also the getPhoneByExternalReferenceCode method? Because I would like to start using it instead of have what we do right now like
```
fetchByERC
if (null) ? throw Exception
```
with the getByERC we should avoid to do every time this check